src/activities.ts

```diff
@@ -1,3 +1,36 @@
-export async function greet(name: string): Promise<string> {
-  return `Hello, ${name}!`;
+import { Context } from '@temporalio/activity'
+import {
+  FulfillmentResult,
+  fulfillmentService,
+  inventoryService,
+  PaymentResult,
+  paymentService,
+  ReservationResult,
+} from './services'
+
+export type Order = {
+  itemId: string
+  quantity: number
+  addressId: string
+  userId: string
+}
+
+const getRequestId = () => Context.current().info.workflowExecution.workflowId
+
+export async function reserve(order: Order): Promise<ReservationResult> {
+  return inventoryService.reserve({ ...order, requestId: getRequestId() })
+}
+export async function unreserve(order: Order): Promise<ReservationResult> {
+  return inventoryService.unreserve({ ...order, requestId: getRequestId() })
+}
+
+export async function charge(order: Order): Promise<PaymentResult> {
+  return paymentService.charge({ ...order, requestId: getRequestId() })
+}
+export async function refund(order: Order): Promise<PaymentResult> {
+  return paymentService.refund({ ...order, requestId: getRequestId() })
+}
```

Comment:
Charge and refund should probably not know anything about inventory, just amount to charge

Reply:
Yeah, for simplicity's sake I think I'll gloss over / hide amount calculation